### DISCUSSION TITLE
Validation with Button

### DISCUSSION THREAD

**Discussion by angelru** [xamarin/XamarinCommunityToolkit#1346] [Q&A]
I am having a hard time validating

I have a custom control with a frame, multiple tags, and an input. I have managed to add the validation for that input, but I can't tell when everything is validated through a command. I have added the click event in the codebehin and there I do the validation, then I run my Command, but all the input values are always null. If in my button I remove the click event and bind the command, everything works fine (without validation), how can I see in my Command if everything went well?

**Comment by vhugogarcia** [comment]
I am facing the same issue. There is no validation at command level.

**Reply by pictos** [reply]
Is the `CanExecute` delegate, in Command ctor helpful?

**Reply by vhugogarcia** [reply]
Hmmm i am not quite sure to follow it. Im sorry. Could you please elaborate more? 😀

**Reply by pictos** [reply]
https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/data-binding/commanding

Here you can find more info about CanExecute
